src/main/java/Parser.java

```diff
@@ -0,0 +1,199 @@
+        try {
+            if (command.toLowerCase().equals("list")) {
+                Ui.printWithBorder(taskList.showList());
+            } else if (commandArray[0].toLowerCase().equals("done")) {
+                if (commandArray.length >= 2) {
+                    markTaskAsDone(taskList, commandArray[1]);
+                } else {
+                    throw new DukeArgumentException("Please specify which task to be marked as done.");
+                }
+            } else if (commandArray[0].toLowerCase().equals("delete")) {
+                if (commandArray.length >= 2) {
+                    deleteTask(taskList, commandArray[1]);
+                } else {
+                    throw new DukeArgumentException("Please specify which task to be deleted.");
+                }
+            } else if (commandArray[0].toLowerCase().equals("find")){
+                if (commandArray.length >= 2) {
+                    findTask(taskList, commandArray[1]);
+                } else {
+                    throw new DukeArgumentException("Please specify the keyword(s) that you want to search.");
+                }
+            } else if (commandArray[0].toLowerCase().equals("todo")) {
+                if (commandArray.length >= 2) {
+                    addTask(taskList, commandArray[1], "T");
+                } else {
+                    throw new DukeArgumentException("Missing field in todo command.");
+                }
+            } else if (commandArray[0].toLowerCase().equals("deadline")) {
+                if (commandArray.length >= 2) {
+                    addTask(taskList, commandArray[1], "D");
+                } else {
+                    throw new DukeArgumentException("Missing field in deadline command.");
+                }
+            } else if (commandArray[0].toLowerCase().equals("event")) {
+                if (commandArray.length >= 2) {
+                    addTask(taskList, commandArray[1], "E");
+                } else {
+                    throw new DukeArgumentException("Missing field in event command.");
+                }
+            } else {
+                throw new DukeUnknownException("Apologies, I do not recognise this command.");
+            }
+        } catch (DukeException err) {
+            Ui.printWithBorder(err.getMessage());
+        }
+    }
+
+    /**
+     * Adds Task to TaskList based on its type (i.e. Todo, Deadline or Event).
+     * @param taskList User's TaskList.
+     * @param input Input provided by the user.
+     * @param type Type of Task ("T" for Todo, "D" for Deadline, "E" for Event).
+     * @throws DukeTaskException
+     */
+    public static void addTask(TaskList taskList, String input, String type) throws DukeTaskException {
+        String str = "\nCurrent number of task(s): ";
+
+        if (type.equals("T")) {
+            Task todo = new Todo(input);
+            taskList.add(todo);
+            Ui.printWithBorder("The following to-do has been added:\n    "
```

Comment:
The printWithBorder function could be refactored further to take in Task and taskList as inputs and print out the required response instead of taking in a string each time.